### ISSUE TITLE
Predicting Complementary Colors Example is stuck in a local minimum

### ISSUE THREAD

**Issue by nanchenchen** [tensorflow/tfjs-core#429]
Hello! I am playing around with the demo examples but I found the Predicting Complementary Colors Example keeps predicting the same value for all inputs. Every iteration the value is different, but all the inputs get the exact same value even after 1000 rounds:
![image](https://user-images.githubusercontent.com/3320040/33845985-09d86532-de5a-11e7-8883-670c5109628b.png)

I did a few attempts to tweak the codes, and then I found if I removed the layer 1 and 2 of the model by commenting out Line 65 ~70,
https://github.com/PAIR-code/deeplearnjs/blob/master/demos/complementary-color-prediction/complementary-color-prediction.ts#L65

I can get different results, which seems to be correct:
![image](https://user-images.githubusercontent.com/3320040/33840914-1f71d86a-de4b-11e7-8a6e-511dab70ecb7.png)

After digging into the issue deeper, I found the model seems to be stuck in a local minimum. 
For the original 3-layer model, I was able to get out of the local minimum by decreasing the batch size to 30 and increasing the learning rate to 0.42 (10x to the original value). With that, I was able to get something more promising in about 600 iterations:
![image](https://user-images.githubusercontent.com/3320040/33846268-0246dbd6-de5b-11e7-9ffb-f6e55b0b9575.png)

And it has a very good result in the end:
![image](https://user-images.githubusercontent.com/3320040/33846294-1bfae7d4-de5b-11e7-88d2-dd09c6106e6e.png)

I was also able to use AdamOptimizer with learning rate=1, beta1=0.9, beta2=0.999, and batch size=300 to get out of the local minimum. 

I think as a demoing example, it may be easier to reduce the example to either single layer (since it is a simple math transformation) or to tweak the learning rate & batch size to get better results faster. Otherwise, I original thought there may be some problems with the library... ;P
